TPP.ArgsParsing.Tests/TypeParsersTest.cs

```diff
@@ -199,6 +199,33 @@ public async Task TestOptionalParser()
             Assert.IsFalse(result3.IsPresent);
         }
 
+        [Test]
+        public async Task TestPercentageParser()
+        {
+            var argsParser = new ArgsParser();
+            argsParser.AddArgumentParser(new PercentageParser());
+
+            Percentage percentageOver100 = await argsParser.Parse<Percentage>(ImmutableList.Create("1234.5678%"));
+            Percentage percentageExactly100 = await argsParser.Parse<Percentage>(ImmutableList.Create("100%"));
+            Percentage percentageZero = await argsParser.Parse<Percentage>(ImmutableList.Create("0%"));
+            Assert.AreEqual(1234.5678, percentageOver100.AsPercent);
+            Assert.AreEqual(12.345678, percentageOver100.AsRatio);
+            Assert.AreEqual(100.0, percentageExactly100.AsPercent);
+            Assert.AreEqual(1.0, percentageExactly100.AsRatio);
+            Assert.AreEqual(0.0, percentageZero.AsPercent);
+            Assert.AreEqual(0.0, percentageZero.AsRatio);
+
+            ArgsParseFailure failureNoFloat = Assert.ThrowsAsync<ArgsParseFailure>(() =>
```

Comment:
I'm somewhat confuse, this (and the error message) refers to the percentage as a float, but internally it's a double. if it's intended to be a user facing response, perhaps decimal is better than either